### ISSUE TITLE
Different rendering on different platforms?

### ISSUE THREAD

**Issue by pastk** [organicmaps/organicmaps#4467]
It looks like the rendering is a bit different across the platforms?
Or is it just difference in dpi? Or a renderer implementation maybe (its GL in both cases).
	
See the screenshots, first one is linux desktop, second is an Android emulator.
The "ladder"/dash pattern is almost invisible on the desktop, but clear on Android (on any zoom level).
55.7128212, 37.5506516 on max zoom

![rend_diff_cablecar](https://user-images.githubusercontent.com/18434508/218333732-8ba01eab-4f3e-42a8-b11f-e8070ef91b67.png)

	
On desktop the difference between powerlines and fence is not as significant as on Android (line widths look much more apart).
55.6249865, 37.6948498 on max zoom

![rend_diff_powerlines](https://user-images.githubusercontent.com/18434508/218333741-e9e6e671-50e3-45ac-acd3-637068709318.png)

My desktop:
```
Renderer = Mesa DRI Intel(R) HD Graphics 5500 (BDW GT2) | Api = OpenGLES3 | Version = 4.6 (Core Profile) Mesa 20.0.8
GLX version: 1.4
OpenGL ES profile version string: OpenGL ES 3.1 Mesa 20.0.8
OpenGL ES profile shading language version string: OpenGL ES GLSL ES 3.10
```

**Comment by vng** [comment]
I suspect that here we have different scale factors. Could you please find output of this log on both devices?
LOG(LINFO, ("Visual scale =", vs, "; Tile size =", tileSize));

**Comment by vng** [comment]
You can find
```
m_ratio = devicePixelRatio();
```
in map_widget.cpp and assign m_ratio manually like 2 or even 3. And check how it looks on your Linux. I suppose that m_ratio = 3 is the best for Retina MacBook laptops.

**Comment by pastk** [comment]
> I suspect that we have different scale factors here. Could you please find output of this log on both devices? LOG(LINFO, ("Visual scale =", vs, "; Tile size =", tileSize));

Linux desktop: `Visual scale = 1 ; Tile size = 512`
Android emulator: `Visual scale = 1.5 ; Tile size = 512`


> ```
> m_ratio = devicePixelRatio();
> ```
> 
> in map_widget.cpp and assign m_ratio manually like 2 or even 3. And check how it looks on your Linux. I suppose that m_ratio = 3 is the best for Retina MacBook laptops.

On setting `3` the aerialway looks better, but fonts are worse and look at the bench icon:
![cablecar_scalefactor3](https://user-images.githubusercontent.com/18434508/219061139-978ece55-3d70-4ab4-9482-e4135de1e3cd.png)

Then on z15 the aerialway becomes dashed (on settings `1.5` and `2` it happens on max zoom also and the fonts are worse even). 

![cablecar_scalefactor3_z15](https://user-images.githubusercontent.com/18434508/219061251-61b44f4d-7038-4d4e-862f-8bfd5be08925.png)

And powerlines are looking even more thin now compared to the fence.

So I didn't find a setting that works for me overall better than the default `1`...

My laptop screen is 13" 1366x768, so its pretty low dpi by nowadays standards.

**Comment by pastk** [comment]
Are there any other ideas how to fix it?

Or we should just amend our styles to look right on low dpi, visual scale = 1 displays?

CC @biodranik

**Comment by vng** [comment]
I don't think that it's worth to make any efforts to make mdpi beauty. Also, Linux GL ES drivers can also influence here.

You can try to play more with constants in visual_params.cpp:

Lets say that male visual scale = 0.8 - 1.5, but keep GetResourcePostfix = mdpi for fonts and symbols.

**Comment by biodranik** [comment]
I would investigate first, if it's a Linux driver issue or not, by testing it on a Mac on the same resolution screen (monitor).

If it's definitely not a driver issue, then the right approach is to apply a smarter selection of an appropriate dpi/visual scale selection on the startup depending on the screen resolution.

**Comment by pastk** [comment]
> I don't think that it's worth making any effort to make mdpi beautiful. Linux GL ES drivers can also influence here.
> 
> You can try to play more with constants in visual_params.cpp:
> 
> Try to make visual scale in range [0.8;1.5], but keep GetResourcePostfix = mdpi for fonts and symbols.

No improvement still. When visual scale != 1.0 the line either gets ugly or with gaps/dashed, i.e. worse than with `1.0`. The ladder pattern starts to be visible from around `2.8` and higher only. Locking the fonts and icons to `mdpi` doesn't improve fonts quality, but makes icons tiny for higher visual scale values. Anyway doesn't make sense to try fix the fonts if we can't make the ladder pattern (the original issue) to be visible.

**Comment by vng** [comment]
I didn't mean to make locking in prod :) This is like a workaround for your laptop.
Make visual scale = 3 for lines, but keep mdpi for fonts and icons.

**Comment by pastk** [comment]
> Make visual scale = 3 for lines, but keep mdpi for fonts and icons.

As I wrote before - it doesn't work good. Not for the lines (becomes "dashed)", not for the fonts...

**Comment by biodranik** [comment]
It looks like a float number rounding issue. Do you use 32bit architecture or 64bit Linux?

**Comment by pastk** [comment]
> It looks like a float number rounding issue. Do you use 32bit architecture or 64bit Linux?

its x64

**Comment by Altonss** [comment]
On a 1920*1080 24'' monitor, OM desktop with antialiasing enabled has still some rendering issue on the powerline:
![image](https://user-images.githubusercontent.com/66519591/228665280-bd609d98-355d-43fd-aef4-d363c08289bb.png)

Compare it to the red line which has no issues...

**Comment by pastk** [comment]
> On a 1920*1080 24'' monitor, OM desktop with antialiasing enabled has still some rendering issue on the powerline: ![image](https://user-images.githubusercontent.com/66519591/228665280-bd609d98-355d-43fd-aef4-d363c08289bb.png)
> 

Thanks! Could you please add an output of `glxinfo -B` and a line from OM log with `Init(): Visual scale =`?

**Comment by Altonss** [comment]
> Thanks! Could you please add an output of `glxinfo -B` and a line from OM log with `Init(): Visual scale =`?

For `glxinfo -B`:

```
name of display: :0
display: :0  screen: 0
direct rendering: Yes
Extended renderer info (GLX_MESA_query_renderer):
    Vendor: Intel (0x8086)
    Device: Mesa Intel(R) UHD Graphics 620 (KBL GT2) (0x5917)
    Version: 22.3.7
    Accelerated: yes
    Video memory: 7865MB
    Unified memory: yes
    Preferred profile: core (0x1)
    Max core profile version: 4.6
    Max compat profile version: 4.6
    Max GLES1 profile version: 1.1
    Max GLES[23] profile version: 3.2
OpenGL vendor string: Intel
OpenGL renderer string: Mesa Intel(R) UHD Graphics 620 (KBL GT2)
OpenGL core profile version string: 4.6 (Core Profile) Mesa 22.3.7
OpenGL core profile shading language version string: 4.60
OpenGL core profile context flags: (none)
OpenGL core profile profile mask: core profile

OpenGL version string: 4.6 (Compatibility Profile) Mesa 22.3.7
OpenGL shading language version string: 4.60
OpenGL context flags: (none)
OpenGL profile mask: compatibility profile

OpenGL ES profile version string: OpenGL ES 3.2 Mesa 22.3.7
OpenGL ES profile shading language version string: OpenGL ES GLSL ES 3.20

```

For `Init(): Visual scale =`:

 ```
Init(): Visual scale = 1 ; Tile size = 1024
```
